relayer/core/relay.go

```diff
@@ -23,8 +24,9 @@ import (
 )
 
 type Relay struct {
-	ethChain chain.Chain
-	subChain chain.Chain
+	subChain    chain.Chain
+	ethSubChain chain.Chain
+	paraChain   chain.Chain
```

Comment:
this, and all the beefy stuff is actually the relayChain, not the paraChain - may want to rename/keep in mind when you refactor the copied files in this PR over to the main substrate folder

subChain is the parachain.

Reply:
ideally i think we want a parachain and relaychain config, both of which use shared substrate libraries/files for things in common, and then seperate files of their own for their respective commitments/messages

Reply:
Refactored in latest